src/realmModels/Observation.js

```diff
@@ -270,11 +270,13 @@ class Observation extends Realm.Object {
   };
 
   static appendObsPhotos = ( obsPhotos, currentObservation ) => {
+    const updatedObs = (currentObservation instanceof Realm.Object
+                        ? currentObservation.toJSON( ) : currentObservation);
+
```

Comment:
This doesn't conform to our eslint rules. Please make sure you are running a linter in your editor and/or have set up [our precommit hooks](https://github.com/inaturalist/iNaturalistReactNative#set-up-pre-commit-hooks), which will run eslint and fix this kind of issue automatically before commit.